hazelcast/src/main/java/com/hazelcast/spi/LiveOperations.java

```diff
@@ -18,83 +18,12 @@
 
 import com.hazelcast.nio.Address;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import static com.hazelcast.util.Preconditions.checkNotNull;
-
 /**
- * Collections per member the callIds of all live operations.
- *
- * The Trace is not thread-safe and is recycled.
- *
- * This data-structure can be optimized to generate less litter. Instead of using an ArrayList for Long objects,
- * use an array for primitive longs. Also the lists don't need to be recreated every time the
- * {@link LiveOperationsTracker#populate(LiveOperations)} method is called; they could be recycled. This would be easy
- * to do since the {@link LiveOperations} is not used by a single thread.
+ * Collects per member the callIds of all live operations.
+ * The object is not thread-safe and is recycled.
  */
-public final class LiveOperations {
-
-    private final Address localAddress;
-    private final Map<Address, List<Long>> callIdsByMember = new HashMap<Address, List<Long>>();
-
-    public LiveOperations(Address localAddress) {
-        this.localAddress = checkNotNull(localAddress, "local address can't be null");
-    }
-
-    public void add(Address address, long callId) {
-        if (callId == 0) {
-            // it is an unregistered operation
-            return;
-        }
-
-        if (address == null) {
-            address = localAddress;
-        }
-
-        List<Long> callIds = callIdsByMember.get(address);
-        if (callIds == null) {
-            callIds = new ArrayList<Long>();
-            callIdsByMember.put(address, callIds);
-        }
-
-        callIds.add(callId);
-    }
-
-    public Set<Address> addresses() {
-        return callIdsByMember.keySet();
-    }
-
-    public long[] callIds(Address address) {
-        List<Long> callIdList = callIdsByMember.get(address);
-        if (callIdList == null) {
-            throw new IllegalArgumentException("unknown address");
-        }
-
-        long[] array = new long[callIdList.size()];
-        for (int k = 0; k < array.length; k++) {
-            array[k] = callIdList.get(k);
-        }
-
-        return array;
-    }
+public interface LiveOperations {
 
-    public void clear() {
-        callIdsByMember.clear();
-    }
+    void add(Address address, long callId);
```

Comment:
some javadoc might be useful here - Saying that this is the caller's address and call id of the operation which is to be tracked.